client/state/purchases/selectors/get-included-domain-transfer.ts

```diff
@@ -0,0 +1,31 @@
+import { isDomainTransfer } from '@automattic/calypso-products';
+import { isSubscription } from 'calypso/lib/purchases';
+import { getSitePurchases } from './get-site-purchases';
+import type { Purchase } from 'calypso/lib/purchases/types';
+import type { AppState } from 'calypso/types';
+import 'calypso/state/purchases/init';
+
+/**
+ * Returns a purchase object that corresponds to that subscription's transferred domain
+ *
+ * Even if a domain transfer was purchased with the subscription, it will
+ * not be returned if the domain transfer product was paid for separately (eg: if it was
+ * renewed on its own).
+ * @param   {AppState} state  global state
+ * @param   {Purchase | null | undefined} subscriptionPurchase  subscription purchase object
+ * @returns {Purchase | null | undefined} domain transfer purchase if there is one, null if none found or not a subscription object passed
+ */
+export const getIncludedDomainTransfer = (
+	state: AppState,
+	subscriptionPurchase: Purchase | null | undefined
+): Purchase | null | undefined => {
+	if ( ! subscriptionPurchase || ! isSubscription( subscriptionPurchase ) ) {
```

Comment:
Should this check `subscriptionPurchase.includedDomainPurchaseAmount` also, the same way that `getIncludedDomainPurchase` does?

It's a fair amount harder to imagine a paid domain transfer being bundled together with a plan than it is for a domain registration (given that a domain transfer can't be renewed) but I think there might be some ways it could happen.  And for consistency it would probably be best for this to behave the same as `getIncludedDomainPurchase` anyway.  So I think I'd lean towards doing it.

Reply:
I'll trust your judgement as I have very little knowledge into this field. I'll make the changes as discussed.

Reply:
Fixed in [ff339ae](https://github.com/Automattic/wp-calypso/pull/102997/commits/ff339aec1b0ffef7d911a4f44fcca2c28ed5f398)